### PR TITLE
DRY tests with `Kirby\TestCase`

### PR DESCRIPTION
**Pull request by distantnative**

### CODE REVIEW

File: tests/TestCase.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		}
+	}
+
+	/**
+	 * Checks if the test class extending this test case class
+	 * has defined a temporary directory
+	 */
+	protected function hasTmp(): bool
+	{
+		return defined(get_class($this) . '::TMP');
```
Comment (lukasbestle) [inline tests/TestCase.php]: Reading this again myself, I think this can be:

```suggestion
		return defined(static::class . '::TMP');
```

**Review by lukasbestle — COMMENTED** [review]
Good idea. Can we replace all other uses of `PHPUnit\Framework\TestCase` with this as well to have the methods available throughout the system?

**Comment by distantnative** [comment]
@lukasbestle change it so that all test classes use `Kirby\TestCase`. Added `assertIsPage` in a few places, but I think it's too big to change all tests to use these methods all at once.

**Review by bastianallgeier — DISMISSED** [review]
This is great. I'm really happy that we are finally start cleaning up the tests.

**Review by bastianallgeier — APPROVED** [review]

**Pull request merged**